app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,92 @@
+class TweetsController < ApplicationController
+
+#shows tweets index page, showing all tweets
+  get '/tweets' do
+    if logged_in? #if user is logged in, show all tweets in database via index view
+      @tweets = Tweet.all
+      erb :'/tweets/index'
+    else
+      redirect '/login' #if user is not logged in, show login page
+    end
+  end
+
+#show form to create new tweet
+  get '/tweets/new' do
+    if logged_in?  #if logged in, show the new tweet form via new view
+      erb :'tweets/new'
+    else
+      redirect '/login' #if not logged in, show login screen
+    end
+  end
+
+  #show a specific tweet given its ID
+    get '/tweets/:id' do
+      if logged_in? #if logged in...
+        @tweet = Tweet.find_by_id(params[:id]) #find specific tweet by ID and assign to @tweet
+        erb :'/tweets/show' #show content in @tweet via show view
+      else
+        redirect '/login'
+      end
+    end
+
+#process new tweet form submission based on params
+  post '/tweets' do
+    if logged_in? #if logged in...
+      if !params[:content].empty? #and if tweet_content isn't empty...
+        @tweet = current_user.tweets.build(params) #create new tweet instance via #build
+        @tweet.save #save new tweet instance to database
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.

In this case, we could lose the `if !params[:content].empty? ` statement entirely, add the validation for content in the Tweet model, and use this pattern:
```
if @tweet.save # If @tweet is invalid it won't save
  redirect "/tweets/#{@tweet.id}"
else
  redirect '/tweets/new' 
end
```

This comment also applies to the patch and delete routes.